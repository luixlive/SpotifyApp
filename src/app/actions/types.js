// Device reducer
export const DEVICE_TYPE_CHANGED = 'DEVICE_TYPE_CHANGED';

// User
export const LOAD_USER = 'LOAD_USER';
export const LOAD_USER_FAILED = 'LOAD_USER_FAILED';
export const LOAD_USER_SUCCEEDED = 'LOAD_USER_SUCCEEDED';
export const LOGOUT_USER = 'LOGOUT_USER';
export const LOGOUT_USER_FAILED = 'LOGOUT_USER_FAILED';
export const LOGOUT_USER_SUCCEEDED = 'LOGOUT_USER_SUCCEEDED';

// UserStats
export const LOAD_USER_STATS = 'LOAD_USER_STATS';
export const LOAD_USER_STATS_FAILED = 'LOAD_USER_STATS_FAILED';
export const LOAD_USER_STATS_SUCCEEDED = 'LOAD_USER_STATS_SUCCEEDED';
