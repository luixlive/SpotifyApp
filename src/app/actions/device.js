import { DEVICE_TYPE_CHANGED } from './types';

export default deviceMobile => (
  { type: DEVICE_TYPE_CHANGED, payload: { deviceMobile } }
);
