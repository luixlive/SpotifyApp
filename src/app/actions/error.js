import { CLEAR_ERROR } from './types';

export default () => (
  { type: CLEAR_ERROR, payload: {} }
);
