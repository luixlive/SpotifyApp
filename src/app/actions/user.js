import { LOAD_USER, LOGOUT_USER } from './types';

export const loadUser = () => (
  { type: LOAD_USER, payload: { } }
);

export const logoutUser = () => (
  { type: LOGOUT_USER, payload: { } }
);
