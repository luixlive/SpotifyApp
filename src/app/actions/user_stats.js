import { LOAD_USER_STATS } from './types';

export default () => (
  { type: LOAD_USER_STATS, payload: { } }
);
