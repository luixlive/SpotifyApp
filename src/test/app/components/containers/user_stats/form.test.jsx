import _ from 'lodash';
import configureStore from 'redux-mock-store';
import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import React from 'react';
import renderer from 'react-test-renderer';

import initialState from './../../../../test_utils/initial_state';
import * as types from './../../../../../app/actions/types';
import Form, {
  PureForm,
} from './../../../../../app/components/containers/user_stats/form';
import {
  form as componentProps,
} from './../../../../test_utils/components_props';

describe('App Components - Form', () => {
  describe('Snapshots', () => {
    it('renders', () => {
      const props = _.cloneDeep(componentProps);
      const rendered = renderer.create(<PureForm {...props} />).toJSON();
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('Behavior', () => {
    let props;
    beforeAll(() => {
      props = _.cloneDeep(componentProps);
    });

    it('renders the values for the current type', () => {
      props.artistsTimeRange = 'medium_term';
      const wrapper = mount(<PureForm {...props} />);
      expect(wrapper.props().type).toEqual('tracks');
      expect(wrapper.find('Radio').get(0).props.checked).toBeTruthy();
      expect(wrapper.find('Radio').get(1).props.checked).toBeFalsy();
      expect(wrapper.find('Radio').get(2).props.checked).toBeFalsy();

      wrapper.setProps({ type: 'artists' });
      expect(wrapper.props().type).toEqual('artists');
      expect(wrapper.find('Radio').get(0).props.checked).toBeFalsy();
      expect(wrapper.find('Radio').get(1).props.checked).toBeTruthy();
      expect(wrapper.find('Radio').get(2).props.checked).toBeFalsy();
    });

    it('marks checked values as false if type is unknown', () => {
      props.type = 'unknown';
      const wrapper = shallow(<PureForm {...props} />);
      expect(wrapper.find('Radio').get(0).props.checked).toBeFalsy();
      expect(wrapper.find('Radio').get(1).props.checked).toBeFalsy();
      expect(wrapper.find('Radio').get(2).props.checked).toBeFalsy();
    });

    it('returns null if an item is clicked but type is unknown', () => {
      props.type = 'unknown';
      const wrapper = shallow(<PureForm {...props} />);
      expect(wrapper.instance().handleChange()()).toBeNull();
    });
  });

  describe('Provider', () => {
    const mockStore = configureStore();
    let store;
    let wrapper;
    beforeAll(() => {
      store = mockStore(initialState);
      wrapper = mount((
        <Provider store={store}>
          <Form type="tracks" />
        </Provider>
      ));
    });

    it('renders', () => {
      expect(wrapper.find(PureForm).length).toEqual(1);
    });

    it('matches initial state', () => {
      expect(Object.keys(wrapper.find(PureForm).props()))
        .toEqual(Object.keys(componentProps));
    });

    it('dispatches changeTracksTimeRange', () => {
      expect(store.getActions().length).toBe(0);
      wrapper.find('Radio').at(1).simulate('change');

      let expectedAction = {
        type: types.CHANGE_TRACKS_TIME_RANGE,
        payload: { timeRange: 'medium_term' },
      };
      expect(store.getActions()[0]).toEqual(expectedAction);

      wrapper.find('Radio').at(2).simulate('change');

      expectedAction = {
        type: types.CHANGE_TRACKS_TIME_RANGE,
        payload: { timeRange: 'long_term' },
      };
      expect(store.getActions()[1]).toEqual(expectedAction);

      store = mockStore(_.merge({}, initialState, {
        userStats: { topTracks: { timeRange: 'long_term' } },
      }));
      wrapper = mount((
        <Provider store={store}>
          <Form type="tracks" />
        </Provider>
      ));
      wrapper.find('Radio').at(0).simulate('change');

      expectedAction = {
        type: types.CHANGE_TRACKS_TIME_RANGE,
        payload: { timeRange: 'short_term' },
      };
      expect(store.getActions()[0]).toEqual(expectedAction);
    });
  });
});
