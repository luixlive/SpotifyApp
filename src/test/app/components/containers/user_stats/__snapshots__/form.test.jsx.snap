// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Components - Form Snapshots renders 1`] = `
<div
  className="ui green segment"
>
  <form
    action={undefined}
    className="ui form"
    onSubmit={[Function]}
  >
    <h3
      className="ui header"
    >
      Since
    </h3>
    <div
      className="grouped fields"
    >
      <div
        className="field"
      >
        <div
          className="ui checked radio checkbox"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={true}
            className="hidden"
            id={undefined}
            name="timeRange"
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="radio"
            value={undefined}
          />
          <label
            htmlFor={undefined}
          >
            last month
          </label>
        </div>
      </div>
      <div
        className="field"
      >
        <div
          className="ui radio checkbox"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            id={undefined}
            name="timeRange"
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="radio"
            value={undefined}
          />
          <label
            htmlFor={undefined}
          >
            last six months
          </label>
        </div>
      </div>
      <div
        className="field"
      >
        <div
          className="ui radio checkbox"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            id={undefined}
            name="timeRange"
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="radio"
            value={undefined}
          />
          <label
            htmlFor={undefined}
          >
            beginning of times
          </label>
        </div>
      </div>
    </div>
  </form>
</div>
`;
